pydantic/dataclasses.py

```diff
@@ -201,6 +201,17 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
 
         original_cls = cls
 
+        # we warn on conflicting config specifications, but only if the class doesn't have a dataclass base
+        # because a dataclass base might provide a __pydantic_config__ attribute that we don't want to warn about
+        has_dataclass_base = cls.__bases__ != (object,)
+        if config and (getattr(cls, '__pydantic_config__', None) and not has_dataclass_base):
+            warn(
+                f'`config` is set via both the `dataclass` decorator and `__pydantic_config__` for dataclass {cls.__name__}. '
+                f'This is not supported. The `config` specification from `dataclass` decorator will take priority.',
```

Comment:
```suggestion
                f'The `config` specification from `dataclass` decorator will take priority.',
```
I'd say theoretically it is supported, just that one takes the priority over the other. wdyt?